trieste/models/gpflow/interface.py

```diff
@@ -142,13 +142,15 @@ def log(self) -> None:
         summary_writer = logging.get_tensorboard_writer()
         if summary_writer:
             with summary_writer.as_default(step=logging.get_step_number()):
-                logging.scalar("kernel.variance", self.get_kernel().variance)
-                lengthscales = self.get_kernel().lengthscales
-                if tf.rank(lengthscales) == 0:
-                    logging.scalar("kernel.lengthscale", lengthscales)
-                elif tf.rank(lengthscales) == 1:
-                    for i, lengthscale in enumerate(lengthscales):
-                        logging.scalar(f"kernel.lengthscale[{i}]", lengthscale)
+                kernel = self.get_kernel()
+                if isinstance(kernel, gpflow.kernels.Stationary):
```

Comment:
hard to say if this would be a good enough check, I don't know GPflow well enough - perhaps add a test for this?

it might be good enough to support only the basic case, but I think we probably can do something more generic, like fetching all trainable variables? I think they also come with names so we can still make it descriptive enough, no?

Reply:
The check should be sufficient: gpflow.kernels.Stationary is where the lengthscales and variance parameters are set.

Makes sense to add a test. I'll see how much we can extract generically, but it might be tricky inferring types, shapes, etc. If it's too much work this may have to do for now (though I can add it to a WIBNI issue).